datadog_checks_base/tests/test_time.py

```diff
@@ -0,0 +1,60 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from datetime import datetime
+
+import pytest
+from dateutil import tz
+from six import PY2
+
+from datadog_checks.base.utils.time import EPOCH, UTC, get_current_datetime, get_timestamp, normalize_datetime
+
+pytestmark = pytest.mark.time
+
+
+def test_timestamp_type():
+    assert isinstance(get_timestamp(), float)
+
+
+class TestNormalization:
+    def test_replace(self):
+        now = datetime.now()
+        assert now.tzinfo is None
+
+        assert normalize_datetime(now).tzinfo is UTC
+
+    def test_utc(self):
+        nyc = tz.gettz('America/New_York')
+        now = datetime.now(nyc)
+
+        assert normalize_datetime(now).tzinfo is nyc
+
+
+class TestCurrentDatetime:
+    def test_default_utc(self):
+        dt = get_current_datetime()
+        now = datetime.now(UTC)
+
+        assert now.tzinfo is UTC
+        assert get_timestamp(now) - get_timestamp(dt) < 0.01
+
+    def test_tz(self):
+        nyc = tz.gettz('America/New_York')
+        dt = get_current_datetime(nyc)
+        now = datetime.now(nyc)
+
+        assert now.tzinfo is nyc
+        assert get_timestamp(now) - get_timestamp(dt) < 0.01
+
+
+@pytest.mark.skipif(PY2, reason='Using Python 3 features')
+class TestConstants:
+    def test_epoch(self):
+        from datetime import timezone
+
+        assert EPOCH == datetime(1970, 1, 1, tzinfo=timezone.utc)
+
+    def test_utc(self):
+        from datetime import timezone
+
+        assert UTC.utcoffset(None) == timezone.utc.utcoffset(None)
```

Comment:
What is the purpose of this test?

Is it because you're exposing a public UTC object and it's an implementation detail that you're exposing it directly from `dateutil`?

Reply:
I'm trying to assert that the constant we expose is a tzinfo instance and represents UTC. How would you do that?

Reply:
This is fine I think. I was just surprised because `UTC` is already tested upstream, so it's already part of the `dateutil` contract that it's UTC. I think it's fair to write your own test for it, though, considering you see this as part of the public interface (and the fact that it's supplied directly from `dateutil.tz` is an implementation detail.

That said, if in the future you drop Python 2 and change it from `UTC = dateutil.tz.UTC` to `UTC = datetime.timezone.utc`, would you still want this test?

Reply:
Aha, that's an interesting question. Yes when that happens I will indeed remove this test. Therefore, I will remove it now since the contract is the same (tested upstream). Good call.